code/datums/mutations/sight.dm

```diff
@@ -42,20 +42,36 @@
 	text_gain_indication = "<span class='notice'>You can see the heat rising off of your skin...</span>"
 	time_coeff = 2
 	instability = 25
-	var/visionflag = TRAIT_THERMAL_VISION
+	power = /obj/effect/proc_holder/spell/self/thermal_vision_activate
 
-/datum/mutation/human/thermal/on_acquiring(mob/living/carbon/human/owner)
-	if(..())
+/obj/effect/proc_holder/spell/self/thermal_vision_activate
+	name = "Activate thermal vision"
+	desc = "You can see thermal signatures, at the cost of your eyesight."
+	charge_max = 20 SECONDS
+	clothes_req = FALSE
+
+/obj/effect/proc_holder/spell/self/thermal_vision_activate/cast(list/targets, mob/user = usr)
+	. = ..()
+
+	if(HAS_TRAIT(user,TRAIT_THERMAL_VISION))
 		return
 
-	ADD_TRAIT(owner, visionflag, GENETIC_MUTATION)
-	owner.update_sight()
+	ADD_TRAIT(user, TRAIT_THERMAL_VISION, GENETIC_MUTATION)
+	user.update_sight()
+	to_chat(user, text("You focus your eyes intensely, as your vision becomes filled with heat signatures."))
 
-/datum/mutation/human/thermal/on_losing(mob/living/carbon/human/owner)
-	if(..())
+	addtimer(CALLBACK(src, .proc/thermal_vision_deactivate), 10 SECONDS)
```

Comment:
if you lose your thermal vision mutation/spell before this procs, do you still lose your thermal vision?

Reply:
Good point, I'll add a check.